### ISSUE TITLE
CTS test asserts

### ISSUE THREAD

**Issue by ziga-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8978]
On ubuntu with RADV the test `dEQP-VK.wsi.display_control.register_device_event` asserts:

> Test case 'dEQP-VK.wsi.display_control.register_device_event'..
> deqp-vk: /home/nnn/Projects/Vulkan-ValidationLayers/layers/./thread_tracker/thread_safety_validation.h:116: std::shared_ptr<ObjectUseData> counter<T>::FindObject(T, const Location&) [with T = VkFence_T*]: Assertion `object_table.contains(object)' failed.
> Aborted (core dumped)

**Comment by spencer-lunarg** [comment]
@ziga-lunarg is this still an issue?

**Comment by ziga-lunarg** [comment]
Yes I can still reproduce (`Thread Safety` needs to be enabled).

Test case 'dEQP-VK.wsi.display_control.register_device_event'..
deqp-vk: /home/nnn/Projects/Vulkan-ValidationLayers/layers/./thread_tracker/thread_safety_validation.h:122: std::shared_ptr<threadsafety::ObjectUseData> threadsafety::Counter<T>::FindObject(T, const Location&) [with T = VkFence_T*]: Assertion `object_table.contains(object)' failed.
Aborted (core dumped)

**Comment by spencer-lunarg** [comment]
yep, can reproduce....will try and figure out

**Comment by spencer-lunarg** [comment]
found issue, the test calls `vkDestroyFence` and the fence handle was added via `vkRegisterDeviceEventEXT` 

We have a thing in ObjectTracker for `vkRegisterDeviceEvent`/`vkRegisterDisplayEvent` I remember adding, seems I just need to add it to Thread Safety layer as well
